modules/local_storage/src/LocalStockChecker.php

```diff
@@ -3,42 +3,40 @@
 namespace Drupal\commerce_stock_local;
 
 use Drupal\commerce_stock\StockCheckInterface;
-use Drupal\commerce_stock\StockUpdateInterface;
+use Drupal\Core\Database\Connection;
+use Symfony\Component\DependencyInjection\ContainerInterface;
 
-/**
- * The backend for the local stock service.
- */
-class LocalStockStorage implements StockCheckInterface, StockUpdateInterface {
+class LocalStockChecker implements StockCheckInterface {
 
   /**
-   * {@inheritdoc}
+   * The database connection.
+   *
+   * @var \Drupal\Core\Database\Connection
    */
-  public function createTransaction($entity_id, $location_id, $zone, $quantity, $unit_cost, $transaction_type_id, array $metadata) {
-    // Get optional fields.
-    $related_tid = isset($metadata['related_tid']) ? $metadata['related_tid'] : NULL;
-    $related_oid = isset($metadata['related_oid']) ? $metadata['related_oid'] : NULL;
-    $related_uid = isset($metadata['related_uid']) ? $metadata['related_uid'] : NULL;
-    $data = isset($metadata['data']) ? $metadata['data'] : NULL;
+  protected $database;
 
-    // Create a record.
-    $field_values = [
-      'entity_id' => $entity_id,
-      'qty' => $quantity,
-      'location_id' => $location_id,
-      'location_zone' => $zone,
-      'unit_cost' => $unit_cost,
-      'transaction_time' => time(),
-      'transaction_type_id' => $transaction_type_id,
-      'related_tid' => $related_tid,
-      'related_oid' => $related_oid,
-      'related_uid' => $related_uid,
-      'data' => serialize($data),
-    ];
-    $insert = \Drupal::database()->insert('commerce_stock_transaction')
-      ->fields(array_keys($field_values))
-      ->values(array_values($field_values))->execute();
+  /**
+   * Constructs the local stock checker.
+   *
+   * @param \Drupal\Core\Database\Connection $database
+   *   The database connection.
+   */
+  public function __construct(Connection $database) {
```

Comment:
I suppose its a good thing having the database dependency injected but how will this work in real life. I can see why you may want to hold the stock data in a database that's not the main drupal DB but how can you control what database the commerce_stock_local_schema() in the install file will target ? 
Not at all worried but curies how it would work